experiment/simulation/js/main.js

```diff
@@ -1 +1,416 @@
+        easing: 'linear',
+    }).add({
+        delay: '1000',
+        opacity: 1
+    }).add({
+        update: function(anim) {
+            message = messages[5];
+            document.getElementById("message").innerHTML = message;
+        }
+    });
+}
+
+function dropbead(){
+    const animation_1 = anime.timeline({
+        targets:'#bead',
+        duration: 1200, 
+        easing: 'linear',
+    }).add({
+        translateY: '7.6vw',
+    }).add({
+        update: function(anim) {
+            message = messages[6];
+            document.getElementById("message").innerHTML = message;
+        }
+    });
+}
+
+function stir_bead(){
+    const animation_1 = anime.timeline({
+        targets: "#bead", 
+        duration: 1000,
+        easing : 'linear',
+        loop: true,
+        direction: 'alternate',
+    });
+
+    animation_1.add({
+        width : '0.4vw',
+        rotateY: '10',
+        translateX: '0.4vw'
+    });
+
+    const animation_2 = anime.timeline({
+        targets: "#water-flask", 
+        duration: 3000,
+        easing : 'linear',
+    }).add({
+        update: function(anim) {
+            message = messages[8];
+            document.getElementById("message").innerHTML = message;
+        }
+    }).add({
+        delay: '3000',
+        update: function(anim) {
+            message = messages[9];
+            document.getElementById("message").innerHTML = message;
+        }
+    });
+
+}
+
+
+function heating() {
+    knob_heater_flag += 1;
+    if(knob_heater_flag === 2) {
+        const animation_2 = anime.timeline({
+            targets: "#pipette", 
+            duration: 500,
+            easing : 'linear',
+        }).add({
+            delay: '500',
+            opacity: 0.9,
+        }).add({
+            update: function(anim) {
+                message = messages[10];
+                document.getElementById("message").innerHTML = message;
+            }
+        });
+    }
+    else if(knob_heater_flag != 2)
+    {
+        const animation_1 = anime.timeline({
+            targets: '.message',
+            duration: 1000,
+            easing: 'linear'
+        }).add({
+            update: function(anim) {
+                message = messages[7];
+                document.getElementById("message").innerHTML = message;
+            }
+        });
+    }
+}
+
+function drop_pipette_to_beaker() {
+
+    console.log('pipette flag= ');
+    console.log(pipette_flag);
+    if(pipette_flag === 0)
+    {
+        const animation_1 = anime.timeline({
+            targets: "#pipette", 
+            duration: 1000,
+            easing : 'linear',
+        }).add({
+            translateY: '6vw',
+            translateX: '-14vw',
+        }).add({
+            update: function(anim) {
+                message = messages[11];
+                document.getElementById("message").innerHTML = message;
+            }
+        });
+        pipette_flag = 2;
+    }
+}
+
+function pipette_fill() {
```

Comment:
It'd be better to use one encapsulating function for all pipette transitions. This function will have a switch for the pipette_flag and then will call the respective functions based on the pipette_flag value. This way:
1. All transitions will be in one place for anyone to see
2. Functions will not need individual checks for flag value and increments, this stuff can be dealt with in the switch case. That way these functions will be independent of that control code and will focus on their functionality

Reply:
Added comments and explanations